Packs/Citrix/Integrations/CitrixCloudEventCollector/CitrixCloudEventCollector.py

```diff
@@ -0,0 +1,205 @@
+            params["Limit"] = limit
+
+        headers = {
+            "Authorization": f"CwsAuth Bearer={access_token}",
+            "accept": "application/json",
+            "Citrix-CustomerId": self.customer_id,
+        }
+
+        demisto.info(f"Sending http request to get records with {params=}")
+        response = self._http_request(
+            "get", url_suffix="systemlog/records", headers=headers, params=params, ok_codes=[200, 401], resp_type="response"
+        )
+
+        if response.status_code == 401:
+            demisto.info("Access token expired; refreshing...")
+            access_token = self.request_access_token()
+            headers["Authorization"] = f"CwsAuth Bearer={access_token}"
+            demisto.info(f"Sending http request to get records with {params=}")
+
+            return self._http_request("get", url_suffix="systemlog/records", headers=headers, params=params)
+        else:
+            return response.json()
+
+    def get_records_with_pagination(self, limit: int, start_date_time: str | None, end_date_time: str | None = None):
+        records: list[dict] = []
+        continuation_token = None
+        raw_res = None
+
+        while len(records) < int(limit):
+            raw_res = self.get_records(
+                start_date_time=start_date_time, end_date_time=end_date_time, continuation_token=continuation_token, limit=limit
+            )
+            records.extend(raw_res.get("items", []))
+            continuation_token = raw_res.get("continuationToken")
+
+            if not continuation_token:
+                break
+
+        for record in records:
+            record["_time"] = record.get("utcTimestamp")
+
+        return records[:limit], raw_res
+
+
+""" HELPER FUNCTIONS """
+
+
+def get_events_command(client: Client, args: dict):  # type: ignore
+    limit = int(args.get("limit", "100"))
+    start_date_time = args.get("start_date_time")
+    end_date_time = args.get("end_date_time")
+    should_push_events = argToBoolean(args.get("should_push_events", False))
+
+    demisto.debug(f"Running citrix-cloud-get-events with {should_push_events=}")
+
+    records, raw_res = client.get_records_with_pagination(
+        limit=limit, start_date_time=start_date_time, end_date_time=end_date_time
+    )
+
+    results = CommandResults(
+        outputs_prefix="CitrixCloud.Event",
+        outputs_key_field="recordId",
+        outputs=records,
+        readable_output=tableToMarkdown("Events List", records),
+        raw_response=raw_res,
+    )
+
+    if should_push_events:
+        demisto.debug(f"send {len(records)} events to xsiam")
+        send_events_to_xsiam(records, vendor=VENDOR, product=PRODUCT)
+
+    return results
+
+
+def fetch_events_command(client: Client, max_fetch: int, last_run: dict):
+    records, _ = client.get_records_with_pagination(limit=max_fetch, start_date_time=last_run.get("LastRun"))
+
+    if records:
+        last_run = {"LastRun": records[0]["_time"]}
```

Comment:
Shouldnt this be according to the last record? If this is sorted asc lets comment it.

Reply:
The response sort data in descending order, the first value is the latest date
added comment